src/Plugin.php

```diff
@@ -11,22 +11,222 @@
 namespace Dealerdirect\Composer\Plugin\Installers\PHPCodeSniffer;
 
 use Composer\Composer;
+use Composer\EventDispatcher\EventSubscriberInterface;
 use Composer\IO\IOInterface;
+use Composer\Package\AliasPackage;
+use Composer\Package\PackageInterface;
 use Composer\Plugin\PluginInterface;
+use Composer\Script\ScriptEvents;
+use Symfony\Component\Process\Exception\LogicException;
+use Symfony\Component\Process\Exception\ProcessFailedException;
+use Symfony\Component\Process\Exception\RuntimeException;
+use Symfony\Component\Process\Process;
 
 /**
  * PHP_CodeSniffer standard installation manager.
  *
  * @author Franck Nijhof <f.nijhof@dealerdirect.nl>
  */
-class Plugin implements PluginInterface
+class Plugin implements PluginInterface, EventSubscriberInterface
 {
+
+    const PACKAGE_TYPE = 'phpcodesniffer-standard';
+
+    /**
+     * @var Composer
+     */
+    protected $composer;
```

Comment:
If you do not plan on having the `Plugin` class extended, these properties might as well be `private`.